force-app/felles-kodeverk/KodeverkBatchable.cls

```
@@ -0,0 +1,43 @@
+/**
+ * Transforms each meaning in a batch to a common code, then upserts the
+ * batch
+ */
+public class KodeverkBatchable implements Database.Batchable<Common_Code__c> {
+
+    private List<Common_Code__c> codes;
+
+    /**
+     * boilerplate
+     */
+    public KodeverkBatchable(List<Common_Code__c> codes) {
+        this.codes = codes;
+    }
+
+    /**
+     * boilerplate
+     */
+    public Iterable<Common_Code__c> start(Database.BatchableContext context) {
+        return this.codes;
+    }
+
+    /**
+     * the heavy work is done here. transform the responses into records and
+     * upserts them.
+     */
+    public void execute(Database.BatchableContext context, List<Common_Code__c> codes) {
+        try {
+            upsert codes;
+        } catch (DmlException de) {
+            new LoggerUtility('Kodeverk').errorAndPublish(
+                'DmlException occurred when attempting to execute KodeverkBatchable:\n' +
+                de.getMessage()
+            );
+        }
```

Skulle vi hatt en pokemon catch her som vi logger?
For: Vi kan fange og logge med "riktig" domain dersom noe går på tverke og det ikke bare "failing silently" en gang i fremtiden.
Mot: Salesforce kaster visstnok en event når en batch tryner som vi kan fange og logge. Da ville vi fanget "alle" feilende batcher og får logget de for å gi synlighet.

---

som diskutert i går, vi ser det an